pkg/operator/ceph/object/zone/controller.go

```diff
@@ -151,22 +160,18 @@ func (r *ReconcileObjectZone) reconcile(request reconcile.Request) (reconcile.Re
 		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
 		//
 		if !cephObjectZone.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Remove finalizer
+			err := opcontroller.RemoveFinalizer(r.opManagerContext, r.client, cephObjectZone)
+			if err != nil {
+				return reconcile.Result{}, *cephObjectZone, errors.Wrap(err, "failed to remove finalizer")
+			}
 			// Return and do not requeue. Successful deletion.
 			return reconcile.Result{}, *cephObjectZone, nil
 		}
 		return reconcileResponse, *cephObjectZone, nil
 	}
 	r.clusterSpec = &cephCluster.Spec
 
-	// DELETE: the CR was deleted
-	if !cephObjectZone.GetDeletionTimestamp().IsZero() {
-		logger.Debugf("deleting zone CR %q", cephObjectZone.Name)
-		r.recorder.Eventf(cephObjectZone, v1.EventTypeNormal, string(cephv1.ReconcileStarted), "deleting CephObjectZOne %q", cephObjectZone.Name)
-
-		// Return and do not requeue. Successful deletion.
-		return reconcile.Result{}, *cephObjectZone, nil
-	}
-
```

Comment:
Why was this moved so much lower? I understand why it might need to be after `LoadClusterInfo()`, but why after updating Status, etc?

Reply:
@BlaineEXE : I just moved below `zonegroupchecks` in reconcile function, to make sure zonegroup exists before performing the deletion. If required I can change the location of updatestatus() to below the deletion

Reply:
If the zone is being deleted, the zone group must exist? What if the zone group is also be deleted? It doesn't need to exist in that case, right? Or is that needed so reconcileCephZoneGroup() will delete it in that case?

Reply:
@travisn To delete a zone, zonegroup needs to exist IMO. As of now Rook is not deleting `zonegroup/realm` from the backend, just add the finaliser when they request to remove them.
But if it is missing then IMO we can remove the Zone CR without having these checks, otherwise, radosgw-admin commands will fail.
After completing this PR, I plan to work on ZoneGroup/Realm deletion.
In ODF 4.12 point of view, we will support multiple zones in a single zonegroup and realm(most common use case in RHCS as well). So this PR plan to focus only in zone deletion

Reply:
I don't quite see that the code follows that logic though. By that logic, if the code were to find that the zonegroup doesn't exist, it should just remove the finalizer and potentially pools, right?

Reply:
@thotz Why would the zone group not exist though? If the zone has been created in a previous reconcile, the zone group would have also been created. So the missing zone group seems like a corner case that we don't need to worry about. Is there any rook workflow that could result in the zone group not existing when the zone is being deleted? If not, we don't need to support the corner case of it not existing. If the only way a zone group could be deleted is from the toolbox, we don't need to support that. If the toolbox is used and causes corner cases, it's generally out of the Rook support.

Reply:
@travisn  Yeah it is out of current multisite workflow in Rook, so not planning to handle that corner case. I was trying to explain the first question about why zonegroup exisitence is needed before zone deletion

Reply:
Okay, so we are agreed that we'll move this back to where it was and not handle the corner case?

Reply:
@BlaineEXE @travisn Moved above to `r.updateStatus()`, below `r.reconcileObjectZoneGroup()` function fetches realm name from zonegroup CR which is required for `radosgw-admin` commands

Reply:
I see, `reconcileObjectZoneGroup()` doesn't actually reconcile anything, it just retrieves the zone group. Would you mind renaming that method to something like `getObjectZoneGroup()`?

There could be a corner case where the zone group doesn't exist or was deleted and this would fail, but it seems like a corner case not to focus on now.